src/test/operators/join_index_test.cpp

```diff
@@ -130,8 +130,7 @@ class JoinIndexTest : public BaseTest {
       _table_wrapper_m, _table_wrapper_n;
 };
 
-typedef ::testing::Types<AdaptiveRadixTreeIndex, CompositeGroupKeyIndex, BTreeIndex /* , GroupKeyIndex */>
-    DerivedIndices;
+typedef ::testing::Types<AdaptiveRadixTreeIndex, CompositeGroupKeyIndex, BTreeIndex, GroupKeyIndex> DerivedIndices;
```

Comment:
I just realized that the index tests do not cover NULL values. Scrolling through the code, I worry that this PR is breaking more than it is fixing. For example, wouldn't a scan on `WHERE a > 5` also include all NULL values? Same for the join? If this is the case and it still survives the tests, the tests are insufficient.

Maybe we should discuss this in person together with @Bensk1?

Reply:
Your description sounds similar to this issue https://github.com/hyrise/hyrise/issues/670

I also checked  `index_scan_test` and `index_join_test` and couldn't find any null tests.
The only test which takes null values into account is the referenced one below, but it is disabled and even if we enable it, the used table `resources/test_data/tbl/join_operators/int_outer_join_null.tbl`  is not available anymore. https://github.com/hyrise/hyrise/blob/1a2397cdb9297231efa8b44b5693ee04feae7b80/src/test/operators/join_index_test.cpp#L262-L266